stellargraph/layer/rgcn.py

```diff
@@ -0,0 +1,554 @@
+                for i in range(N_edge_types)
+            ]
+
+        # Otherwise, create dense matrices from input tensor
+        else:
+            Ainput = [Lambda(lambda A: K.squeeze(A, 0))(A_) for A_ in As]
+
+        h_layer = x_in
+
+        for layer in self._layers:
+            if isinstance(layer, RelationalGraphConvolution):
+                # For an RGCN layer add the adjacency matrices and output indices
+                # Note that the output indices are only used if `final_layer=True`
+                h_layer = layer([h_layer, out_indices] + Ainput)
+            else:
+                # For other (non-graph) layers only supply the input tensor
+                h_layer = layer(h_layer)
+
+        return h_layer
+
+    def node_model(self):
+        """
+        Builds a RGCN model for node prediction
+
+        Returns:
+            tuple: `(x_inp, x_out)`, where
+            `x_inp` is a list of Keras input tensors for the RGCN model (containing node features,
+             node indices, and the indices and values for the sparse adjacency matrices for each relationship),
+            and `x_out` is a Keras tensor for the RGCN model output.
+        """
+        # Placeholder for node features
+        N_nodes = self.generator.features.shape[0]
+        N_feat = self.generator.features.shape[1]
+        N_edge_types = len(self.generator.As)
+
+        # Inputs for features & target indices
+        x_t = Input(batch_shape=(1, N_nodes, N_feat))
+        out_indices_t = Input(batch_shape=(1, None), dtype="int32")
+
+        # Create inputs for sparse or dense matrices
+        if self.use_sparse:
+            # Placeholders for the sparse adjacency matrix
+            A_indices_t = [
+                Input(batch_shape=(1, None, 2), dtype="int64")
+                for i in range(N_edge_types)
+            ]
+            A_values_t = [Input(batch_shape=(1, None)) for i in range(N_edge_types)]
+            A_placeholders = A_indices_t + A_values_t
+
+        else:
+            # Placeholders for the dense adjacency matrix
+            A_placeholders = [
+                Input(batch_shape=(1, N_nodes, N_nodes)) for i in range(N_edge_types)
+            ]
+
+        x_inp = [x_t, out_indices_t] + A_placeholders
+        x_out = self(x_inp)
+
+        # Flatten output by removing singleton batch dimension
+        if x_out.shape[0] == 1:
+            self.x_out_flat = Lambda(lambda x: K.squeeze(x, 0))(x_out)
+        else:
+            self.x_out_flat = x_out
+
+        return x_inp, x_out
+
+    def build(self):
+        """
+        Builds a RGCN model for node prediction. Link/node pair prediction will added in the future.
+
+        Returns:
+            tuple: (x_inp, x_out), where ``x_inp`` is a list of Keras input tensors
+            for the specified RGCN model and ``x_out`` contains
+            model output tensor(s) of shape (batch_size, layer_sizes[-1])
+
+        """
+
+        if self.generator is not None:
```

Comment:
Look at https://github.com/stellargraph/stellargraph/pull/543 and change this to use the `multiplicity` flag to match the new-style full batch generators (don't implement link prediction yet though, but it would be a good idea to make a new issue to do this)

Reply:
done!